frontend/src/metabase-lib/lib/Question.js

```diff
@@ -290,6 +290,15 @@ export default class Question {
         return Mode.forQuestion(this);
     }
 
+    // see if the card can be xrayed
+    canXray() {
```

Comment:
What is the reasoning behind adding `canXray` and `xrayConst` to question API instead of treating them as global settings that are accessed via settings selectors?

I don't feel very good about the mutation of card object in `initializeQB` unless in know that in very near term those settings become question-specific. 

If the reason is that ClickAction API lets you only use `Question` object as a source of information, I think that we should include global settings in `settings` object or similar to that API, so that we can write stuff like
```
export default ({ question, settings }: ClickActionProps): ClickAction[] => {
   if (settings.canXray) {
      ...
   }
}
```

Reply:
@attekei Yeah the main reason I tried this approach was because I didn't want to have to pollute the `ActionWidget` component with settings and it seemed like the pattern was to have different actions determine whether they're applicable. I also thought that it might make sense to have those properties on questions but I guess if we're going to do that that should just come from the backend.

I think for now I'll try your suggestion of including access to settings in the actions.